internal/appsec/trace/httptrace/http.go

```diff
@@ -100,20 +117,27 @@ func SetSecurityEventsTags(span trace.TagSetter, events []any) {
 }
 
 func init() {
-	collectedHTTPHeaders = make(map[string]struct{}, len(defaultIPHeaders)+len(defaultNonIPHeaders)+1)
-	for _, h := range defaultIPHeaders {
-		collectedHTTPHeaders[h] = struct{}{}
-	}
-	for _, h := range defaultNonIPHeaders {
-		collectedHTTPHeaders[h] = struct{}{}
+	makeCollectedHTTPHeadersLookupMap()
+	readMonitoredClientIPHeadersConfig()
+}
+
+func makeCollectedHTTPHeadersLookupMap() {
+	collectedHeadersLookupMap = make(map[string]struct{}, len(defaultCollectedHeaders))
+	for _, h := range defaultCollectedHeaders {
+		collectedHeadersLookupMap[h] = struct{}{}
 	}
+}
 
-	if cfg := os.Getenv(envClientIPHeader); cfg != "" {
-		// Add this header to the list of collected headers
-		collectedHTTPHeaders[cfg] = struct{}{}
+func readMonitoredClientIPHeadersConfig() {
+	if header := os.Getenv(envClientIPHeader); header != "" {
 		// Make this header the only one to consider in ClientIP
-		monitoredClientIPHeadersCfg = []string{cfg}
+		monitoredClientIPHeadersCfg = []string{header}
+
+		// Add this header to the list of collected headers
+		header = normalizeHTTPHeaderName(header)
+		collectedHeadersLookupMap[header] = struct{}{}
 	} else {
+		// No specific IP header was configured, use the default list
 		monitoredClientIPHeadersCfg = defaultIPHeaders
```

Comment:
I am wondering if we could make this env var support a list of headers comma-separated to get the same result as `defaultIPHeaders` variable. Anyway just an idea

Reply:
to do indeed upon user request ;-)